specutils/spectra/xrayspectrum1d.py

```diff
@@ -0,0 +1,573 @@
+import logging
+
+import numpy as np
+from astropy import units as u
+from astropy.io import fits
+from .spectrum1d import Spectrum1D
+
+__all__ = ['XraySpectrum1D', 'AreaResponse', 'ResponseMatrix']
+
+# For dealing with varied unit string choices
+EV   = ['eV', 'ev']
+KEV  = ['kev', 'keV']
+ANGS = ['angs', 'Angs', 'Angstrom', 'angstrom', 'Angstroms', 'angstroms', 'A', 'a']
+
+## An introduction to X-ray analysis can be found here:
+## http://cxc.cfa.harvard.edu/xrayschool/talks/intro_xray_analysis.pdf
+
+def _unit_parser(unit_string):
+    """
+    A function for matching unit strings to the correct Astropy unit
+    """
+    if unit_string in EV:
+        return u.eV
+    if unit_string in KEV:
+        return u.keV
+    if unit_string in ANGS:
+        return u.angstrom
+
+class XraySpectrum1D(Spectrum1D):
+    """
+    Spectrum properties specific to X-ray data
+
+    Parameters
+    ----------
+    bin_lo : numpy.ndarray
+        The left edges for bin values
+
+    bin_hi : numpy.ndarray
+        The right edges for bin values
+
+    bin_unit : string OR astropy.units.Unit
+        Unit for the bin values
+
+    counts : astropy.Quantity array
+        Counts histogram for the X-ray spectrum
+
+    exposure : float
+        Exposure time for the dataset
+
+    arf : specutils.AreaResponse or string, default None
+        Strings will be passed to AreaResponse.__init__
+        All other input types are stored as arf attribute
+
+    rmf : specutils.ResponseMatrix or string, default None
+        Strings will be passed to ResponseMatrix.__init__
+        All other input types are stored as rmf attribute
+
+    rest_value : astropy.units.Quantity, default 0 Angstrom
+        See Spectrum1D rest_value input
```

Comment:
Again, perfect for 
```
See `~specutils.spectra.Spectrum1D`
```
(or whereever Spectrum1D is defined. I don't know on top of my head).